serialize/encoder.go

```diff
@@ -52,12 +52,17 @@ import (
 var (
 	byteOrder        binary.ByteOrder = binary.LittleEndian
 	headerMagicBytes                  = encodeUInt16(headerMagicNumber)
+
+	// FOLLOWUP(prateek): this terrible hack is necessary due to a bug in the indexing
+	// sub-system. Our FST library (Vellum) doesn't handle '' values correctly in an edge
+	// case. I'll remove this once I have some cycles to come back to it.
+	mapEmptyTagValueToSpaceHack = ident.BytesID([]byte(" "))
 )
 
 var (
-	errTagEncoderInUse   = errors.New("encoder already in use")
-	errTagLiteralTooLong = errors.New("literal is too long")
-	errEmptyTagLiteral   = errors.New("tag nam/value cannot be empty")
+	errTagEncoderInUse     = errors.New("encoder already in use")
+	errTagLiteralTooLong   = errors.New("literal is too long")
+	errEmptyTagNameLiteral = errors.New("tag name cannot be empty")
```

Comment:
Can you wrap this in an invalid params error (to return Bad Request)?  `xerrors.NewInvalidParamsError(..)`?

Reply:
sure thing